examples/tests/limitedclient.rs

```diff
@@ -0,0 +1,55 @@
+//! Check that `limitedclient` is successful in its goal of not linking in any
+//! AES code.
+//!
+//! We also check `simpleclient` (which includes everything) as a baseline to
+//! detect errors in the test code.
+
+// Don't assume binutils are available everywhere, or that `nm` has a
+// portable interface.
+#![cfg(target_os = "linux")]
+
+use std::process::Command;
+
+#[test]
+fn simpleclient_contains_aes_symbols() {
+    assert!(count_aes_symbols_in_executable(env!("CARGO_BIN_EXE_simpleclient")) > 0);
+}
+
+#[test]
+fn limitedclient_does_not_contain_aes_symbols() {
+    let limitedclient = env!("CARGO_BIN_EXE_limitedclient");
+    if fips_mode(limitedclient) {
+        println!("FIPS mode includes the entirety of the fipsmodule due to dynamic linking");
+        return;
+    }
+    assert_eq!(count_aes_symbols_in_executable(limitedclient), 0);
+}
+
+fn fips_mode(exe: &str) -> bool {
+    symbols_in_executable(exe)
+        .lines()
+        .any(|sym| sym.starts_with("aws_lc_fips_"))
+}
+
+fn count_aes_symbols_in_executable(exe: &str) -> usize {
+    symbols_in_executable(exe)
+        .lines()
+        .inspect(|sym| println!("candidate symbol {sym:?}"))
+        .filter(|sym| sym.starts_with("aws_lc_") && sym.ends_with("_EVP_aead_aes_128_gcm_tls13"))
+        .inspect(|sym| println!("found aes symbol {sym:?}"))
+        .count()
```

Comment:
IMO this would be clearer as a `for ln in symbols_in_executable(exe).lines()` loop.